src/Security/src/Authorization.Certificate/ApplicationInstanceCertificate.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.RegularExpressions;
+
+namespace Steeltoe.Security.Authorization.Certificate;
+
+internal sealed class ApplicationInstanceCertificate
+{
+    // This pattern is found on certificates issued by Diego
+    private const string CloudFoundryInstanceCertificateSubjectRegex =
+        @"^CN=(?<instance>[0-9a-f-]+),\sOU=organization:(?<org>[0-9a-f-]+)\s\+\sOU=space:(?<space>[0-9a-f-]+)\s\+\sOU=app:(?<app>[0-9a-f-]+)$";
+
+    // This pattern is found on certificates created in Steeltoe
+    private const string SteeltoeInstanceCertificateSubjectRegex =
+        @"^CN=(?<instance>[0-9a-f-]+),\sOU=app:(?<app>[0-9a-f-]+)\s\+\sOU=space:(?<space>[0-9a-f-]+)\s\+\sOU=organization:(?<org>[0-9a-f-]+)$";
+
+    public string OrganizationId { get; private set; }
+
+    public string SpaceId { get; private set; }
+
+    public string ApplicationId { get; private set; }
+
+    public string InstanceId { get; private set; }
+
+    private ApplicationInstanceCertificate(string organizationId, string spaceId, string applicationId, string instanceId)
+    {
+        OrganizationId = organizationId;
+        SpaceId = spaceId;
+        ApplicationId = applicationId;
+        InstanceId = instanceId;
+    }
+
+    public static bool TryParse(X509Certificate2 certificate, [MaybeNullWhen(false)] out ApplicationInstanceCertificate outInstanceCertificate, ILogger logger)
+    {
+        outInstanceCertificate = null;
+
+        Match instanceMatch = Regex.Match(certificate.Subject.Replace("\"", string.Empty, StringComparison.Ordinal),
```

Comment:
There's no need to run the same replacement twice.

Reply:
Thanks for pointing that out... As it turns out, all of the tests pass without running the replacement at all

Reply:
I did not check the history, I was pretty sure I'm the only person that has worked on this area of the code since it was contributed by another party and I have no recollection of this detail. The simplest way to account for more cases is probably to change the signature to accept just the subject as a string rather than require the full certificate (that we aren't actually using anyway) so that we can much more easily test whatever cases we like

Reply:
Sounds good.

Reply:
Sorry, didn't mean to imply I was dismissing the issue of the quotes entirely, rather specifying that changing the signature allows focusing this method on just the subject and making testing of _any_ variation easier

Reply:
Removing the quote replacement had no effect because, while quotes are allowed in subjects in general, they aren't created in either scenario this codebase was actually built to support. I don't think there's any loss if we didn't do the replacement, but I already did the work to refactor the signature and add dedicated test data, so might as well leave it in now.